docs/spec/auth/vesting.md

```diff
@@ -0,0 +1,88 @@
+## Vesting
+
+### Intro and Requirements
+
+This paper specifies changes to the auth and bank modules to implement vesting accounts for the Cosmos Hub. 
+The requirements for this vesting account is that it should be capable of being initialized during genesis with
+a starting balance X coins and a vesting blocktime T. The owner of this account should be able to delegate to validators 
+and vote with locked coins, however they cannot send locked coins to other accounts until those coins have been unlocked. 
+The vesting account should also be able to spend any coins it receives from other users or from fees/inflation rewards. 
+Thus, the bank module's `MsgSend` handler should error if a vesting account is trying to send an amount that exceeds their 
+unlocked coin amount.
+
+### Implementation
+
+##### Vesting Account implementation
+
+```go
+type VestingAccount interface {
+    Account
+    AssertIsVestingAccount() // existence implies that account is vesting.
```

Comment:
Do you think `ConvertAccount() BaseAccount` should also be part of the interface?

Reply:
This function was used to ensure that an account was a vesting account after deserializing from store. 

I think the introduction of `TotalUnlockedCoins` removes needs for this method, since there shouldn't be a non-vesting account that implements that method. Thus, we can assume an account implementing this method is necessarily a vesting account and handle it accordingly. Thoughts?

Reply:
nvm, refactoring `TotalUnlockedCoins` to `SendableCoins`.

It's possible in the future to have non-vesting accounts to implement this method for a different purpose. So having the assert method could clearly distinguish vesting accounts. Not sure if we need to worry about that now though